Userland/Libraries/LibJS/Runtime/ArrayBufferPrototype.cpp

```diff
@@ -37,51 +38,96 @@ ArrayBufferPrototype::~ArrayBufferPrototype()
 // 25.1.5.3 ArrayBuffer.prototype.slice ( start, end ), https://tc39.es/ecma262/#sec-arraybuffer.prototype.slice
 JS_DEFINE_NATIVE_FUNCTION(ArrayBufferPrototype::slice)
 {
+    // 1. Let O be the this value.
+    // 2. Perform ? RequireInternalSlot(O, [[ArrayBufferData]]).
     auto* array_buffer_object = TRY(typed_this_value(global_object));
-    // FIXME: Check for shared buffer
+
+    // 3. If IsSharedArrayBuffer(O) is true, throw a TypeError exception.
+    // fixme: shared buffer
+
+    // 4. If IsDetachedBuffer(O) is true, throw a TypeError exception.
     if (array_buffer_object->is_detached())
         return vm.throw_completion<TypeError>(global_object, ErrorType::DetachedArrayBuffer);
 
+    // 5. Let len be O.[[ArrayBufferByteLength]].
     auto length = array_buffer_object->byte_length();
 
+    // 6. Let relativeStart be ? ToIntegerOrInfinity(start).
     auto relative_start = TRY(vm.argument(0).to_integer_or_infinity(global_object));
 
     double first;
-    if (relative_start < 0)
+    if (Value(relative_start).is_negative_infinity()) {
+        // 7. If relativeStart is -∞, let first be 0.
```

Comment:
This goes above the if

Reply:
I'll amend that - I wasn't a hundred on this, but followed what TypedArrayPrototype was doing :)